### PR TITLE
Fix [Android]Text Clipping Issue in Entry with Long Strings 

### LINKED ISSUE #23774: [Android] Text Clipping Issue in Entry with Long Strings
### Description

When the text string has many characters, the text gets clipped when IsReadOnly is set to true, and we cannot view the first part of the text. Is there any possibility to view the text from the beginning?

https://github.com/user-attachments/assets/1e4d7a54-bf1d-4630-a81a-0375d8d5c489

### Steps to Reproduce

Obtain the sample from the below repository
Replicate the issue as did like in the above video.

### Link to public reproduction project repository

https://github.com/HariharanChokkalingam/EntryIssue

### Version with bug

8.0.70 SR7

### Is this a regression from previous behavior?

No, this is something new

### Last version that worked well

Unknown/Other

### Affected platforms

Android

### Affected platform versions

_No response_

### Did you find any workaround?

No

### Relevant log output

```shell
No Response.
```

### PR DESCRIPTION
**Pull request by devanathan-vaithiyanathan**
### Root Cause
The issue occurs because when IsReadOnly is set to true, the EditText on Android does not reset its text position to the beginning, causing it to display the end of the text instead.

### Description of Change

This change explicitly sets the text selection to the start  when IsReadOnly is true, ensuring that the text will be displayed from the beginning instead of the end.

### Issues Fixed

Fixes #23774 

### Validated the behaviour in the following platforms
 
- [x] Android
- [x] Windows
- [x] iOS
- [x] Mac

### Output Screenshot
| Before  | After  |
|---------|--------|
| **Android**<br>  |**Android**<br>  |

### CODE REVIEW

File: src/Controls/tests/TestCases.Shared.Tests/Tests/Issues/Issue23774.cs [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+        public Issue23774(TestDevice device) : base(device)
+        {
+        }
+
+        [Test]
+		[Category(UITestCategories.Entry)]
+		public void VerifyEntryIsReadOnlyInitialLoading()
+        {
+			App.WaitForElement("Label");
+			VerifyScreenshot();
```
Comment (jsuarezruiz) [inline src/Controls/tests/TestCases.Shared.Tests/Tests/Issues/Issue23774.cs:22]: Test snapshots already available on the latest PR build.

Comment (devanathan-vaithiyanathan) [reply src/Controls/tests/TestCases.Shared.Tests/Tests/Issues/Issue23774.cs:22]: I’ve added snapshots for all platforms

**Comment by jfversluis** [comment]
As mentioned [here](https://github.com/dotnet/maui/issues/23054#issuecomment-2494285174), I don't think we should want to do this. This seems to be the default behavior for Android and I don't see why we would want to change that. If anyone wants to implement this, they can use the code from this PR to make that happen for their project.

**Pull request closed without merging**